File: app/src/main/java/com/gas/app/ui/main/mvp/MainPresenter.java
package com.gas.app.ui.main.mvp;

import android.app.Application;

import com.base.lib.di.scope.ActivityScope;
import com.base.lib.mvp.BasePresenter;
import com.base.lib.mvp.IModel;

import javax.inject.Inject;

import io.reactivex.disposables.Disposable;

@ActivityScope
public class MainPresenter extends BasePresenter<IModel, MainContract.View> {

    @Inject
    Application mApplication;

    private Disposable mDisposable;

    @Inject
    public MainPresenter(MainContract.View rootView) {
        super(rootView);
    }

    @Override
    public void onDestroy() {

    }

}

File: app/src/main/java/com/gas/app/ui/activity/main/mvp/MainPresenter.kt
package com.gas.app.ui.activity.main.mvp

import com.base.lib.di.scope.ActivityScope
import com.base.lib.mvp.BasePresenter

import javax.inject.Inject


/**
 * ================================================
 * Description:
 * <p>
 * Created by GasMvpTemplate on 07/22/2020 20:05
 * ================================================
 */

@ActivityScope
class MainPresenter
@Inject
constructor(model: MainContract.Model, rootView: MainContract.View) : BasePresenter<MainContract.Model, MainContract.View>(model, rootView) {

    override fun onDestroy() {
        super.onDestroy();
    }
}
